### ISSUE TITLE
Support for provisioning Proton resources in CloudFormation

### ISSUE THREAD

**Issue by clareliguori** [aws/aws-proton-public-roadmap#22]
Support for AWS Proton resource types in CloudFormation (for example, AWS::Proton::Environment)

**Comment by ngamradt-turner** [comment]
Hello @clareliguori and @rafavallina, is there an update on this support?  Starting to look into Proton a bit, but really isn't an option for us until it has full IaC support.

**Comment by rafavallina** [comment]
@ngamradt-turner thanks for reaching out! No updates to give, but we are actively working on delivering this functionality. Will post here as we make further progress. A detail that would help with decision making: if we were to deliver some resource types before others (for instance, environments and services before environment templates) - what would be your preferred priority?

**Comment by ngamradt-turner** [comment]
Hello @rafavallina, based on my limited knowledge of the product, it appears that environments and services would be more important since templates can be uploaded as a bundle and reused (this is based on what I have read, haven't worked with it yet).  If I am misunderstanding, please let me know.

**Comment by rafavallina** [comment]
@ngamradt-turner No, I think you got it right. We are thinking about the same. Will continue to update this channel as we make progress!

**Comment by ngamradt-turner** [comment]
Hello @rafavallina, just checking in to see if there are any updates on the CloudFormation support for Proton?

**Comment by rafavallina** [comment]
Hi @ngamradt-turner we are planning to release support for two resources shortly: Environment Account Connections and Environments. We are going to be tackling this one by one. Do you have any specific type of resource that is critical for you?

**Comment by ngamradt-turner** [comment]
Hello @rafavallina, I guess for me this has been a bit of a "chicken-and-egg" situation.  I haven't really worked with Proton because there is no CloudFormation support, but because I haven't worked with Proton, I cannot really say which features are most important.  I suppose I will try to work with Proton a bit when I have some time.

**Comment by rafavallina** [comment]
Yeah, I can see that to be the case. I welcome any feedback that you have as you try it, but I do want to make clear that we will indeed release CFN support for all the types of resources that Proton creates. It's a matter of timing and prioritizing which ones to release first. So, as you experiment with it, feel free to comment here which resources are more critical so we can focus on those!

**Comment by ngamradt-turner** [comment]
Hello @rafavallina, I will put in a research item for myself to work with Proton outside of CloudFormation so that I can provide some useful feedback.

That being said, is active development taking place for CloudFormation support for any Proton resources?

**Comment by rafavallina** [comment]
Yes, absolutely. Without disclosing dates, I'd expect to see the first two Proton resources available in CloudFormation quite soon, and we will continue delivering the rest afterwards. But this is actively in development and progressing well

**Comment by IngussNeilands** [comment]
Hi, any update on this ?

**Comment by tatcoo** [comment]
Hi Inguss - we're making progress on this (despite the delay from the comment in 2021), stay tuned!
